libsolidity/ast/AST_accept.h

```diff
@@ -403,7 +403,7 @@ void Block::accept(ASTVisitor& _visitor)
 void Block::accept(ASTConstVisitor& _visitor) const
 {
 	if (_visitor.visit(*this))
-		listAccept(m_statements, _visitor);
+		if (m_statements.size() > 0 && m_statements[0] != nullptr) listAccept(m_statements, _visitor);
```

Comment:
I think `m_statements[0] != nullptr` doesn't need to be tested here.

Reply:
When there are errors m_statements[0] is null. 

I believe when you change this and run the tests you should see that.

Reply:
Yes, but shouldn't that be caught by the other change (in `listAccept`)?
Also, is it possible that `[0] != nullptr` but `[1] == nullptr` (or `[i] . i >= 1`)?

Reply:
What happens is we because of the "predictive" nature of the parser a statement is created but has an error so it is null. 

The change to listAccept tolerates this. It doesn't throw an error and I don't think we want it to.